dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/PreheatUtils.java

```diff
@@ -0,0 +1,39 @@
+package org.hisp.dhis.tracker.preheat;
+
+import java.util.Optional;
+
+import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.tracker.TrackerIdScheme;
+import org.hisp.dhis.tracker.TrackerIdentifier;
+
+import lombok.experimental.UtilityClass;
+
+/**
+ * @author Luciano Fiandesio
+ */
+@UtilityClass
+public class PreheatUtils
+{
+    public <T extends IdentifiableObject> Optional<String> resolveKey( TrackerIdentifier identifier, T object )
+    {
+        if ( identifier.getIdScheme().equals( TrackerIdScheme.UID ) )
+        {
+            return Optional.ofNullable( object.getUid() );
+        }
+        else if ( identifier.getIdScheme().equals( TrackerIdScheme.CODE ) )
+        {
+            return Optional.ofNullable( object.getCode() );
+        }
+        else if ( identifier.getIdScheme().equals( TrackerIdScheme.NAME ) )
+        {
+            return Optional.ofNullable( object.getName() );
+        }
+        else if ( identifier.getIdScheme().equals( TrackerIdScheme.ATTRIBUTE ) )
+        {
+            return Optional.ofNullable( identifier.getIdentifier( object ) );
+        }
+        // TODO TrackerIdScheme.AUTO ??
```

Comment:
Not sure if this was missed, or will stay here for future implementation?

Reply:
This is more a reminder, because I'm not sure how we should tread `AUTO`: I don't think it is even used now.